src/coreclr/vm/method.cpp

```diff
@@ -3384,11 +3384,22 @@ BOOL NDirectMethodDesc::TryResolveNDirectTargetForNoGCTransition(_In_ MethodDesc
     }
     CONTRACTL_END
 
+    _ASSERTE(pMD->IsNDirect());
+    NDirectMethodDesc* pNMD = reinterpret_cast<NDirectMethodDesc*>(pMD);
+
+    if (!pMD->NDirectTargetIsImportThunk())
+    {
+        // This is an early out to handle already resolved targets
+        *ndirectTarget = pNMD->GetNDirectTarget();
+
+        _ASSERTE(*ndirectTarget == ResolveAndSetNDirectTarget(pNMD));
```

Comment:
```suggestion
```
I do not think this assert is going to be reliable (also, it very expensive).

Reply:
Fixed.